umap/static/umap/js/components/alerts/alert.js

```diff
@@ -0,0 +1,177 @@
+class uMapAlert extends HTMLElement {
+  static info(message, duration = 5000) {
+    const event = new CustomEvent('umap:alert', {
+      bubbles: true,
+      cancelable: true,
+      detail: { message, duration },
+    })
+    document.dispatchEvent(event)
+  }
+
+  // biome-ignore lint/style/useNumberNamespace: Number.Infinity returns undefined by default
+  static error(message, duration = Infinity) {
+    const event = new CustomEvent('umap:alert', {
+      bubbles: true,
+      cancelable: true,
+      detail: { level: 'error', message, duration },
+    })
+    document.dispatchEvent(event)
+  }
+
+  constructor() {
+    super()
+    this.container = this.querySelector('[role="dialog"]')
+    this.element = this.container.querySelector('[role="alert"]')
+  }
+
+  _hide() {
+    this.setAttribute('hidden', 'hidden')
+  }
+
+  _show() {
+    this.removeAttribute('hidden')
+  }
+
+  _displayAlert(detail) {
+    const { level = 'info', duration = 5000, message = '' } = detail
+    this.container.dataset.level = level
+    this.container.dataset.duration = duration
+    this.element.textContent = message
+    this._show()
+    if (Number.isFinite(duration)) {
+      setTimeout(() => {
+        this._hide()
+      }, duration)
+    }
+  }
+
+  connectedCallback() {
+    this.addEventListener('click', (event) => {
```

Comment:
can't we call super instead of duplicating this line for each class ?

Reply:
Yes, let's split it as a dedicated method.